source/visionEnhancementProviders/screenCurtain.py

```diff
@@ -306,8 +306,10 @@ def isScreenFullyBlack() -> bool:
 	- there is only one colour used in the image
 	- the first pixel of the screen capture is black
 
-	Having to iterate over a large bitmap image in python is slow (~4s depending on machine, number of pixels).
-	Using this method to calculate a histogram in native code takes ~0.4s comparably.
+	Iterating over a large bitmap image in python is slow (~4s depending on machine, number of pixels).

```

Comment:
I'd put the code snippet you used to iterate over the bitmap as an example. It gives a much clearer picture of EXACTLY what you have tried. Perhaps in newer versions of Python there are new and faster ways to execute basic operations on arrays of ctypes data.